sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -719,4 +720,27 @@ object DataSource extends Logging {
     }
     globPath
   }
+
+  /**
+   * Called before writing into a FileFormat based data source to make sure the
+   * supplied schema is not empty.
+   * @param schema
+   */
+  private def hasEmptySchema(schema: StructType): Unit = {
+    def hasEmptySchemaInternal(schema: StructType): Boolean = {
```

Comment:
they should be `verifySchema` and `hasEmptySchema`, depending on their return type.

Reply:
You are right @cloud-fan. Given we are raising the error from the function itself, should i rename it to validateSchema ?

Reply:
@cloud-fan I have gone ahead and changed the top level function name to validateSchema. I have kept the internal function name to be hasEmptySchema. Hopefully it makes sense now.